api/distributedcontext/baggage.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distributedcontext
+
+import (
+	"go.opentelemetry.io/otel/api/core"
+)
+
+type rawBaggageMap map[core.Key]core.Value
+
+type Baggage struct {
+	m rawBaggageMap
+}
+
+type BaggageUpdate struct {
+	DropSingleK core.Key
```

Comment:
This pattern is cloned from the existing code. Using `DropKeys` requires one additional allocation, and there is a desire to avoid all unnecessary allocations.

Reply:
I'm not convinced that is worth it, but okay.